src/configuration.cpp

```diff
@@ -451,20 +453,36 @@ chimera::CompiledConfiguration::CompiledConfiguration(
     // Override individual templates if specified in the configuration.
     if (bindingNode_)
     {
-        if (const YAML::Node classTemplateNode = bindingNode_["class"])
-            bindingDefinition_.class_cpp = Lookup(classTemplateNode);
+        if (const auto node = lookupYAMLNode(bindingNode_, "class", "header"))
+            bindingDefinition_.class_h = Lookup(node);
+
+        if (const auto node = lookupYAMLNode(bindingNode_, "class", "source"))
+            bindingDefinition_.class_cpp = Lookup(node);
```

Comment:
Hmm, I just had a thought.
What if we do the following to support both types?
```suggestion
        if (const auto node = bindingNode_["class"]) {
            if (node.IsScalar()) {
                bindingDefinition_.class_cpp = Lookup(node);
            } else {
                bindingDefinition_.class_h = Lookup(node["header"]);
                bindingDefinition_.class_cpp = Lookup(node["source"]);
            }            
        }
```
Does it work?

Reply:
Yes, this would work (done in https://github.com/personalrobotics/chimera/pull/289/commits/6c66716240bdf16ed99e21ac29cc725659d0dec0).

My only concern, however, is the case that the user only provides a source template using the old syntax and expect no header template is being generated. In this case, the user-provided source template would be likely not compatible with the default header template of Chimera.

If we really want backward compatibility, then we need to add a logic not to generate headers when the old syntax is used by the user. I'm a little bit hesitant to introduce that complex logic.

Reply:
Oh, you raise a very good point.  The easiest way I can think to do this is to create a string sentinel value using a UUID that indicates that the file should not be generated and assign it here:
```
DO_NOT_RENDER = "CHIMERA_NO_RENDER_86c10773-c4e8-4b2f-95b3-137f8e4ea0da"
```
Then we can assign this value in the if-block and maybe check for equality in the `Render()` block?

This opens up a new possibility, which is that someone can set these keys to null if they want to suppress generation of either headers or cpp files for one of these templates.  I am not sure when this would be useful, maybe if they are using chimera in conjunction with another tool?

Reply:
Using a sentinel value sounds good to me, but this might still have the issue with the old syntax though because in the old syntax we don't have a chance to use the sentinel value.



> This opens up a new possibility, which is that someone can also set the template to `null` if they want to suppress generation of both headers and cpp files for one of these templates.

Not sure if I get what you mean. What's the difference between using a sentinel value and setting the template tag with `null`? Do you mean we provide both ways?

Reply:
Never mind, I got the answers from your [code snippet](https://github.com/personalrobotics/chimera/pull/289#discussion_r408625960).